lib/Sema/TypeCheckStmt.cpp

```diff
@@ -1005,7 +1005,7 @@ bool TypeChecker::typeCheckCatchPattern(CatchStmt *S, DeclContext *DC) {
 
 static bool isDiscardableType(Type type) {
   return (type->is<ErrorType>() ||
-          type->isNever() ||
+          type->isUninhabited() ||
```

Comment:
Right. `isUninhabited` should already be responsible for figuring this out. Call sites shouldn't have to do any additional drilling down.